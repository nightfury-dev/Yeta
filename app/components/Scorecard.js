import * as _ from 'lodash';
import React from 'react';
import { ListView } from 'react-native';

import ScorecardHeader from './ScorecardHeader';
import ScorecardRow from './ScorecardRow';
import ScorecardFooter from './ScorecardFooter';


class Scorecard extends React.Component {
    getCourse() {
        const game = this.getGame();
        return _.find(
            this.props.courses,
            (c) => { return c.id === game.course; }
        );
    }

    getGame() {
        return _.find(
            this.props.games,
            (g) => { return g.id === this.props.gameId; }
        );
    }

    renderHeader() {
        const game = this.getGame();
        const players = _.filter(
            this.props.players,
            (p) => { return _.some(game.players, (id) => p.id == id) }
        );
        return <ScorecardHeader players={players} />
    }

    renderRow(rowData) {
        const course = this.getCourse();
        const par = course.pars[rowData.holeNumber - 1];
        return <ScorecardRow
            holeNumber={rowData.holeNumber}
            par={par}
            scores={rowData.scores} />
    }

    renderFooter() {
        const course = this.getCourse();
        const game = this.getGame();

        const players = _.filter(
            this.props.players,
            (p) => { return _.some(game.players, (id) => p.id == id) }
        );

        const totalScores = players.map(player => {
            return game.scores[player.id].reduce(
                (score, totalScore) => score + totalScore,
                0
            );
        });

        return <ScorecardFooter course={course} scores={totalScores} />;
    }

    createRowData() {
        const course = this.getCourse();
        const game = this.getGame();

        const players = _.filter(
            this.props.players,
            (p) => { return _.some(game.players, (id) => p.id == id) }
        );

        return course.pars.map((par, index) => {
            const currentHoleScores = players.map((player) => {
                return game.scores[player.id][index];
            });
            return {
                holeNumber: index + 1,
                scores: currentHoleScores
            };
        });
    }

    render() {
        const ds = new ListView.DataSource({rowHasChanged: (r1, r2) => r1 !== r2});
        const dataSource = ds.cloneWithRows(this.createRowData());
        return (
            <ListView
                dataSource={dataSource}
                renderHeader={this.renderHeader.bind(this)}
                renderRow={this.renderRow.bind(this)}
                renderFooter={this.renderFooter.bind(this)}/>
        );
    }
};

export default Scorecard;
