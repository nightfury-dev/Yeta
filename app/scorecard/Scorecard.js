import * as _ from 'lodash';
import React from 'react';
import { connect } from 'react-redux';
import { ListView } from 'react-native';

import ScorecardHeader from './ScorecardHeader';
import ScorecardRow from './ScorecardRow';
import ScorecardFooter from './ScorecardFooter';


class Scorecard extends React.Component {
  constructor(props) {
    super(props);

    this.renderHeader = this.renderHeader.bind(this);
    this.renderRow = this.renderRow.bind(this);
    this.renderFooter = this.renderFooter.bind(this);
  }

  componentWillMount() {
    this.players = _.values(this.props.game.players);
    this.scores = _.values(this.props.game.scores);
    this.holes = _.values(this.props.game.course.holes);

    this.holeScores = {};
    this.playerScores = {};

    this.scores.forEach((score) => {
      const holenumber = score.hole.holenumber;
      const playerId = score.player.id;

      if (!this.holeScores[holenumber]) {
        this.holeScores[holenumber] = {};
      }
      this.holeScores[holenumber][playerId] = score.score;

      if (!this.playerScores[playerId]) {
        this.playerScores[playerId] = 0;
      }
      this.playerScores[playerId] += score.score;
    });
  }

  createRowData() {
    return this.holes.map((hole) => {
      const currentHoleScores = this.players.map((player) =>
                this.holeScores[hole.holenumber][player.id]
            );
      return {
        holenumber: hole.holenumber,
        par: hole.par,
        scores: currentHoleScores
      };
    });
  }

  renderRow(rowData) {
    return (<ScorecardRow
      holenumber={rowData.holenumber}
      par={rowData.par}
      scores={rowData.scores}
    />);
  }

  renderFooter() {
    const totalScores = this.players.map((player) =>
          this.playerScores[player.id]
        );

    return (<ScorecardFooter
      course={this.props.game.course}
      scores={totalScores}
    />);
  }

  renderHeader() {
    return <ScorecardHeader players={this.players} />;
  }

  render() {
    const ds = new ListView.DataSource(
            { rowHasChanged: (r1, r2) => r1 !== r2 }
        );
    const dataSource = ds.cloneWithRows(this.createRowData());
    return (<ListView
      dataSource={dataSource}
      renderHeader={this.renderHeader}
      renderRow={this.renderRow}
      renderFooter={this.renderFooter}
      enableEmptySections
    />);
  }
}

Scorecard.propTypes = {
  game: React.PropTypes.object.isRequired
};

const mapStateToProps = (state) => ({
  game: state.games.current
});

export default connect(mapStateToProps)(Scorecard);

export {
  Scorecard
};

