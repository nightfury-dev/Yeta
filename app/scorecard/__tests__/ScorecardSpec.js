import * as _ from 'lodash';
import 'react-native';
import React from 'react';

/* eslint import/no-extraneous-dependencies: 0 */
import renderer from 'react-test-renderer';

import { Scorecard } from '../Scorecard';

const gameWithoutScores = {
  id: 1,
  timeBegin: '2016-12-21T17:44:01.251Z',
  players: [
    {
      id: 1,
      name: 'keijo'
    },
    {
      id: 2,
      name: 'Pertti'
    }
  ],
  course: {
    id: 1,
    name: 'Lyhyt rata',
    holes: [
      {
        id: 1,
        holenumber: 1,
        par: 5
      },
      {
        id: 2,
        holenumber: 2,
        par: 5
      }
    ]
  },
  currentHole: 1
};

const scoresEvenPar = [
  {
    id: 1,
    hole: {
      id: 1,
      holenumber: 1,
      par: 5
    },
    player: {
      id: 1,
      name: 'keijo'
    },
    score: 5
  },
  {
    id: 2,
    hole: {
      id: 2,
      holenumber: 2,
      par: 5
    },
    player: {
      id: 1,
      name: 'keijo'
    },
    score: 5
  },
  {
    id: 4,
    hole: {
      id: 1,
      holenumber: 1,
      par: 5
    },
    player: {
      id: 2,
      name: 'Pertti'
    },
    score: 5
  },
  {
    id: 5,
    hole: {
      id: 2,
      holenumber: 2,
      par: 5
    },
    player: {
      id: 2,
      name: 'Pertti'
    },
    score: 5
  }
];

const scoresUnderPar = [
  {
    id: 1,
    hole: {
      id: 1,
      holenumber: 1,
      par: 5
    },
    player: {
      id: 1,
      name: 'keijo'
    },
    score: 4
  },
  {
    id: 2,
    hole: {
      id: 2,
      holenumber: 2,
      par: 5
    },
    player: {
      id: 1,
      name: 'keijo'
    },
    score: 5
  },
  {
    id: 4,
    hole: {
      id: 1,
      holenumber: 1,
      par: 5
    },
    player: {
      id: 2,
      name: 'Pertti'
    },
    score: 5
  },
  {
    id: 5,
    hole: {
      id: 2,
      holenumber: 2,
      par: 5
    },
    player: {
      id: 2,
      name: 'Pertti'
    },
    score: 5
  }
];

const scoresOverPar = [
  {
    id: 1,
    hole: {
      id: 1,
      holenumber: 1,
      par: 5
    },
    player: {
      id: 1,
      name: 'keijo'
    },
    score: 6
  },
  {
    id: 2,
    hole: {
      id: 2,
      holenumber: 2,
      par: 5
    },
    player: {
      id: 1,
      name: 'keijo'
    },
    score: 5
  },
  {
    id: 4,
    hole: {
      id: 1,
      holenumber: 1,
      par: 5
    },
    player: {
      id: 2,
      name: 'Pertti'
    },
    score: 5
  },
  {
    id: 5,
    hole: {
      id: 2,
      holenumber: 2,
      par: 5
    },
    player: {
      id: 2,
      name: 'Pertti'
    },
    score: 5
  }
];


xdescribe('Scorecard renders correctly', () => {
  test('Even par', () => {
    const gameEvenScores = _.cloneDeep(gameWithoutScores);
    gameEvenScores.scores = scoresEvenPar;
    const tree = renderer.create(
      <Scorecard game={gameEvenScores} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Under par', () => {
    const gameUnderPar = _.cloneDeep(gameWithoutScores);
    gameUnderPar.scores = scoresUnderPar;
    const tree = renderer.create(
      <Scorecard game={gameUnderPar} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('Over par', () => {
    const gameOverPar = _.cloneDeep(gameWithoutScores);
    gameOverPar.scores = scoresOverPar;
    const tree = renderer.create(
      <Scorecard game={gameOverPar} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
