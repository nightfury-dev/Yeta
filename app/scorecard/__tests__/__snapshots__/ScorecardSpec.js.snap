exports[`Scorecard renders correctly Even par 1`] = `
<ScrollView
  dataSource={
    ListViewDataSource {
      "items": 2,
    }
  }
  initialListSize={10}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={1000}
  onKeyboardDidHide={undefined}
  onKeyboardDidShow={undefined}
  onKeyboardWillHide={undefined}
  onKeyboardWillShow={undefined}
  onLayout={[Function]}
  onScroll={[Function]}
  pageSize={1}
  removeClippedSubviews={true}
  renderFooter={[Function]}
  renderHeader={[Function]}
  renderRow={[Function]}
  scrollEventThrottle={50}
  scrollRenderAheadDistance={1000}
  stickyHeaderIndices={Array []}
  style={
    Object {
      "backgroundColor": "#3E2723",
      "flex": 1,
      "paddingTop": 64,
    }
  }>
  <View
    style={
      Object {
        "borderBottomWidth": 1,
        "borderColor": "#E65100",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "padding": 5,
      }
    }>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 1,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "color": "#EEE5E9",
            "fontSize": 20,
          }
        }>
        # (par)
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "color": "#EEE5E9",
            "fontSize": 20,
          }
        }>
        keijo
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "color": "#EEE5E9",
            "fontSize": 20,
          }
        }>
        Pertti
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "borderBottomWidth": 1,
        "borderColor": "#E65100",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "padding": 5,
      }
    }>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 1,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "color": "#EEE5E9",
            "fontSize": 20,
          }
        }>
        1 (5)
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "color": "#EEE5E9",
            "fontSize": 20,
          }
        }>
        5
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "color": "#EEE5E9",
            "fontSize": 20,
          }
        }>
        5
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "borderBottomWidth": 1,
        "borderColor": "#E65100",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "padding": 5,
      }
    }>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 1,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "color": "#EEE5E9",
            "fontSize": 20,
          }
        }>
        2 (5)
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "color": "#EEE5E9",
            "fontSize": 20,
          }
        }>
        5
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "color": "#EEE5E9",
            "fontSize": 20,
          }
        }>
        5
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "borderBottomWidth": 1,
        "borderColor": "#E65100",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "padding": 5,
      }
    }>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 1,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "color": "#EEE5E9",
            "fontSize": 20,
          }
        }>
        (10)
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "color": "#EEE5E9",
            "fontSize": 20,
          }
        }>
        10 (0)
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "color": "#EEE5E9",
            "fontSize": 20,
          }
        }>
        10 (0)
      </Text>
    </View>
  </View>
</ScrollView>
`;

exports[`Scorecard renders correctly Over par 1`] = `
<ScrollView
  dataSource={
    ListViewDataSource {
      "items": 2,
    }
  }
  initialListSize={10}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={1000}
  onKeyboardDidHide={undefined}
  onKeyboardDidShow={undefined}
  onKeyboardWillHide={undefined}
  onKeyboardWillShow={undefined}
  onLayout={[Function]}
  onScroll={[Function]}
  pageSize={1}
  removeClippedSubviews={true}
  renderFooter={[Function]}
  renderHeader={[Function]}
  renderRow={[Function]}
  scrollEventThrottle={50}
  scrollRenderAheadDistance={1000}
  stickyHeaderIndices={Array []}
  style={
    Object {
      "backgroundColor": "#3E2723",
      "flex": 1,
      "paddingTop": 64,
    }
  }>
  <View
    style={
      Object {
        "borderBottomWidth": 1,
        "borderColor": "#E65100",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "padding": 5,
      }
    }>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 1,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "color": "#EEE5E9",
            "fontSize": 20,
          }
        }>
        # (par)
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "color": "#EEE5E9",
            "fontSize": 20,
          }
        }>
        keijo
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "color": "#EEE5E9",
            "fontSize": 20,
          }
        }>
        Pertti
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "borderBottomWidth": 1,
        "borderColor": "#E65100",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "padding": 5,
      }
    }>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 1,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "color": "#EEE5E9",
            "fontSize": 20,
          }
        }>
        1 (5)
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "color": "#EEE5E9",
            "fontSize": 20,
          }
        }>
        6
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "color": "#EEE5E9",
            "fontSize": 20,
          }
        }>
        5
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "borderBottomWidth": 1,
        "borderColor": "#E65100",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "padding": 5,
      }
    }>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 1,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "color": "#EEE5E9",
            "fontSize": 20,
          }
        }>
        2 (5)
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "color": "#EEE5E9",
            "fontSize": 20,
          }
        }>
        5
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "color": "#EEE5E9",
            "fontSize": 20,
          }
        }>
        5
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "borderBottomWidth": 1,
        "borderColor": "#E65100",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "padding": 5,
      }
    }>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 1,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "color": "#EEE5E9",
            "fontSize": 20,
          }
        }>
        (10)
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "color": "#EEE5E9",
            "fontSize": 20,
          }
        }>
        11 (+1)
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "color": "#EEE5E9",
            "fontSize": 20,
          }
        }>
        10 (0)
      </Text>
    </View>
  </View>
</ScrollView>
`;

exports[`Scorecard renders correctly Under par 1`] = `
<ScrollView
  dataSource={
    ListViewDataSource {
      "items": 2,
    }
  }
  initialListSize={10}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={1000}
  onKeyboardDidHide={undefined}
  onKeyboardDidShow={undefined}
  onKeyboardWillHide={undefined}
  onKeyboardWillShow={undefined}
  onLayout={[Function]}
  onScroll={[Function]}
  pageSize={1}
  removeClippedSubviews={true}
  renderFooter={[Function]}
  renderHeader={[Function]}
  renderRow={[Function]}
  scrollEventThrottle={50}
  scrollRenderAheadDistance={1000}
  stickyHeaderIndices={Array []}
  style={
    Object {
      "backgroundColor": "#3E2723",
      "flex": 1,
      "paddingTop": 64,
    }
  }>
  <View
    style={
      Object {
        "borderBottomWidth": 1,
        "borderColor": "#E65100",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "padding": 5,
      }
    }>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 1,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "color": "#EEE5E9",
            "fontSize": 20,
          }
        }>
        # (par)
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "color": "#EEE5E9",
            "fontSize": 20,
          }
        }>
        keijo
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "color": "#EEE5E9",
            "fontSize": 20,
          }
        }>
        Pertti
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "borderBottomWidth": 1,
        "borderColor": "#E65100",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "padding": 5,
      }
    }>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 1,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "color": "#EEE5E9",
            "fontSize": 20,
          }
        }>
        1 (5)
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "color": "#EEE5E9",
            "fontSize": 20,
          }
        }>
        4
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "color": "#EEE5E9",
            "fontSize": 20,
          }
        }>
        5
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "borderBottomWidth": 1,
        "borderColor": "#E65100",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "padding": 5,
      }
    }>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 1,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "color": "#EEE5E9",
            "fontSize": 20,
          }
        }>
        2 (5)
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "color": "#EEE5E9",
            "fontSize": 20,
          }
        }>
        5
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "color": "#EEE5E9",
            "fontSize": 20,
          }
        }>
        5
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "borderBottomWidth": 1,
        "borderColor": "#E65100",
        "flex": 1,
        "flexDirection": "row",
        "justifyContent": "space-between",
        "padding": 5,
      }
    }>
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 1,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "color": "#EEE5E9",
            "fontSize": 20,
          }
        }>
        (10)
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "color": "#EEE5E9",
            "fontSize": 20,
          }
        }>
        9 (-1)
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
        }
      }>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Object {
            "color": "#EEE5E9",
            "fontSize": 20,
          }
        }>
        10 (0)
      </Text>
    </View>
  </View>
</ScrollView>
`;
